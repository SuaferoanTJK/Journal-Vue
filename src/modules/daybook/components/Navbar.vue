<script>
import { ref } from "@vue/reactivity";
import { useRouter } from "vue-router";
import useAuth from "@/modules/auth/composables/useAuth";

export default {
  name: "Nav-Component",
  setup() {
    const hidden = ref(true);
    const router = useRouter();
    const { username, logout } = useAuth();

    return {
      hidden,
      username,
      dropdown: () => {
        hidden.value = !hidden.value;
      },
      onLogout: () => {
        router.push({ name: "login" });
        logout();
      },
    };
  },
};
</script>

<template>
  <nav class="bg-white px-4 py-2.5 shadow-md">
    <div
      class="container flex flex-wrap justify-between items-center md:w-10/12 mx-auto"
    >
      <router-link to="/" class="flex items-center">
        <img src="@/assets/logo.png" class="mr-3 h-6 sm:h-9" alt="Vue Logo" />
        <span class="self-center text-xl font-semibold whitespace-nowrap">
          {{ username }}
        </span>
      </router-link>
      <button
        type="button"
        @click="dropdown"
        class="inline-flex items-center p-2 text-sm text-gray-500 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200"
      >
        <svg
          class="w-6 h-6"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            fill-rule="evenodd"
            d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
            clip-rule="evenodd"
          ></path>
        </svg>
      </button>
      <div
        class="justify-between items-center w-full md:flex md:w-auto md:order-1"
        :class="[hidden ? 'hidden' : 'absolute top-10 left-0']"
      >
        <ul
          class="flex flex-col p-4 mt-4 bg-gray-50 rounded-lg border border-gray-100 md:flex-row md:space-x-8 md:mt-0 md:text-sm md:font-medium md:border-0 md:bg-white"
        >
          <li class="flex gap-3 items-center text-slate-700 hover:text-sky-700">
            <button
              @click="onLogout"
              class="block py-2 pr-4 pl-3 rounded md:bg-transparent md:p-0"
            >
              Logout
            </button>
            <i class="fa-solid fa-arrow-right-from-bracket"></i>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</template>
