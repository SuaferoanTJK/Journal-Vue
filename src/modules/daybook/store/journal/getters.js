// export const getter = (state) => {
//     return state;
// }

export const getEntriesByTerm =
  (state) =>
  (term = "") => {
    if (term.length === 0) return state.entries;
    return state.entries.filter((entry) =>
      entry.text.toLowerCase().includes(term.toLocaleLowerCase())
    );
  };
export const getEntryByID = (state) => (id) => {
  const entry = state.entries.find((entry) => entry.id == id);
  if (!entry) return;
  return { ...entry };
};
