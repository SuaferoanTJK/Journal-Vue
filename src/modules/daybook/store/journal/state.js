export default () => ({
  isLoading: true,
  entries: [
    {
      id: new Date().getTime(),
      date: new Date().toDateString(),
      text: "Et exercitation mollit aute cupidatat nostrud culpa occaecat cupidatat dolor enim minim esse. Aute do magna ut consectetur eu mollit. Nostrud proident excepteur esse elit nisi. Occaecat exercitation aliquip laborum cupidatat. Sunt dolor est laboris veniam ipsum. Deserunt elit commodo labore est in cupidatat et dolore.",
      picture: null,
    },
    {
      id: new Date().getTime() + 1000,
      date: new Date().toDateString(),
      text: "Do aute fugiat do qui nostrud nulla qui elit est duis laboris labore ipsum do. Eiusmod quis consectetur reprehenderit aute voluptate elit dolor. Consectetur exercitation duis velit deserunt ipsum amet ad et consequat ut.",
      picture: null,
    },
    {
      id: new Date().getTime() + 2000,
      date: new Date().toDateString(),
      text: "Elit amet Lorem aliquip voluptate deserunt mollit anim duis. Eiusmod voluptate amet et reprehenderit consequat nostrud cupidatat. Magna deserunt consequat est proident exercitation eiusmod qui. Do ad ut proident sint labore ipsum mollit minim sint. Pariatur minim culpa aliquip amet amet id duis id culpa Lorem irure Lorem qui ut. Culpa ea eiusmod consequat exercitation et incididunt velit.",
      picture: null,
    },
  ],
});
