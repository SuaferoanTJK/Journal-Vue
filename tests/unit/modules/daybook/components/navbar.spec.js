import { shallowMount } from "@vue/test-utils";
import { createStore } from "vuex";

import Navbar from "@/modules/daybook/components/Navbar.vue";
import auth from "@/modules/auth/store";
import journal from "@/modules/daybook/store";
import { journalState } from "../../../mock-data/test-journal-state";

const createVuexStore = (authInitState, journalInitState = journalState) =>
  createStore({
    modules: {
      auth: {
        ...auth,
        state: { ...authInitState },
      },
      journal: {
        ...journal,
        state: { ...journalInitState },
      },
    },
  });

describe("Navbar Unit Testing", () => {
  const store = createVuexStore({
    status: "authenticated",
    user: { name: "AndrÃ©s Felipe", email: "andres@gmail.com" },
    idToken: "ABC123",
    refreshToken: "XYZ123",
  });

  test("should display component", () => {
    const wrapper = shallowMount(Navbar, {
      global: {
        plugins: [store],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
