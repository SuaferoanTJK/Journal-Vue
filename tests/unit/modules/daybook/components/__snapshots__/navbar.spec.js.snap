// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Unit Testing should display component 1`] = `
<nav class="bg-white px-4 py-2.5 shadow-md">
  <div class="container flex flex-wrap justify-between items-center md:w-10/12 mx-auto">
    <router-link to="/" class="flex items-center"><img src="" class="mr-3 h-6 sm:h-9" alt="Vue Logo"><span class="self-center text-xl font-semibold whitespace-nowrap">Andr√©s Felipe</span></router-link><button type="button" class="inline-flex items-center p-2 text-sm text-gray-500 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200"><svg class="w-6 h-6" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg">
        <path fill-rule="evenodd" d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z" clip-rule="evenodd"></path>
      </svg></button>
    <div class="justify-between items-center w-full md:flex md:w-auto md:order-1 hidden">
      <ul class="flex flex-col p-4 mt-4 bg-gray-50 rounded-lg border border-gray-100 md:flex-row md:space-x-8 md:mt-0 md:text-sm md:font-medium md:border-0 md:bg-white">
        <li class="flex gap-3 items-center text-slate-700 hover:text-sky-700"><button class="logoutBtn block py-2 pr-4 pl-3 rounded md:bg-transparent md:p-0"> Logout </button><i class="fa-solid fa-arrow-right-from-bracket"></i></li>
      </ul>
    </div>
  </div>
</nav>
`;
